reactor-netty-http/src/test/java/reactor/netty/http/client/WebsocketClientOperationsTest.java

```diff
@@ -48,44 +58,78 @@ private void failOnClientServerError(
-				                                WebsocketServerSpec.builder().protocols(serverSubprotocol).build());
-				                    }))
-				          .wiretap(true)
-				          .bindNow();
+						.port(0)
+						.route(routes ->
+								routes.post("/login", (req, res) -> res.status(serverStatus).sendHeaders())
+										.get("/ws", (req, res) -> {
+											int token = Integer.parseInt(req.requestHeaders().get("Authorization"));
+											if (token >= 400) {
+												return res.status(token).send();
+											}
+											return res.sendWebsocket((i, o) -> o.sendString(Mono.just("test")),
+													WebsocketServerSpec.builder().protocols(serverSubprotocol).build());
+										}))
+						.wiretap(true)
+						.bindNow();
 
 		Flux<String> response =
 				HttpClient.create()
-				          .port(httpServer.port())
-				          .wiretap(true)
-				          .headersWhen(h -> login(httpServer.port()).map(token -> h.set("Authorization", token)))
-				          .websocket(WebsocketClientSpec.builder().protocols(clientSubprotocol).build())
-				          .uri("/ws")
-				          .handle((i, o) -> i.receive().asString())
-				          .log()
-				          .switchIfEmpty(Mono.error(new Exception()));
+						.port(httpServer.port())
+						.wiretap(true)
+						.headersWhen(h -> login(httpServer.port()).map(token -> h.set("Authorization", token)))
+						.websocket(WebsocketClientSpec.builder().protocols(clientSubprotocol).build())
+						.uri("/ws")
+						.handle((i, o) -> i.receive().asString())
+						.log()
+						.switchIfEmpty(Mono.error(new Exception()));
 
 		StepVerifier.create(response)
-		            .expectError(WebSocketHandshakeException.class)
-		            .verify();
+				.expectError(WebSocketHandshakeException.class)
+				.verify();
 
 		httpServer.disposeNow();
 	}
 
 	private Mono<String> login(int port) {
 		return HttpClient.create()
-		                 .port(port)
-		                 .wiretap(true)
-		                 .post()
-		                 .uri("/login")
-		                 .responseSingle((res, buf) -> Mono.just(res.status().code() + ""));
+				.port(port)
+				.wiretap(true)
+				.post()
+				.uri("/login")
+				.responseSingle((res, buf) -> Mono.just(res.status().code() + ""));
+	}
+
+	@Test
+	public void testConfigureWebSocketVersion() {
+		ArgumentCaptor<HttpHeaders> argumentCaptor = ArgumentCaptor.forClass(HttpHeaders.class);
+
+		DisposableServer httpServer = HttpServer.create()
+				.port(0)
+				.handle((in, out) -> {
+					assertThat(in.requestHeaders().get("sec-websocket-version")).isEqualTo("8");
+					return out.sendWebsocket((i, o) -> o.sendString(Mono.just("test")));
+				})
+				.wiretap(true)
+				.bindNow();
+
+		 HttpClient.create()
+				.port(httpServer.port())
+				.wiretap(true)
+				.websocket(WebsocketClientSpec.builder().version(WebSocketVersion.V08).build())
+				.uri("/test")
+				.handle((in, out) -> in.receive().aggregate().asString())
+				.collectList()
+				.block();
```

Comment:
I suggest to specify timeout for the `block` invocation

Reply:
but then if the test fails the error message would be one of timeout... unless you're suggesting to do that in addition to extracting the assertion to perform it on the result of the block?

Reply:
in addition to what you suggested
We always use timeout for `block`, `verify` etc in order to prevent tests hang forever in something happens